### ISSUE TITLE
Support proper power zones from Peloton

### ISSUE THREAD

**Issue by esclee** [philosowaffle/peloton-to-garmin#38]
At the moment, if you don't own a power zone-enabled Garmin device, you have no way of setting your FTP on your GC account; when you get your Peloton workout data into Garmin, the power zone graph will be wildly incorrect. Is there a way to include FTP info on the activities?

**Comment by philosowaffle** [comment]
I'll need to do some more research. I did find this TCX extension that lets you define Heart Rate Zones and Speed Zones.  I'm not sure if Speed Zones are actually the same thing as power zones though.

https://www8.garmin.com/xmlschemas/UserProfileExtensionv1.xsd

**Comment by esclee** [comment]
https://www8.garmin.com/xmlschemas/UserProfilePowerExtensionv1.xsd
This one looks promising? I'm not too familiar with how tcx files are supposed to be structured...

**Comment by philosowaffle** [comment]
Honestly, same, just figuring things out as I go along.  But yes, that looks much closer to the schema we want.  Next things to research:

1. How do we use this correctly? Are we able to include this in each activity tcx file?
1. If not, is there another way to send this data to garmin?

**Comment by clipin** [comment]
I believe https://syncmyworkout.com implemented this. In testing, I see `Workout Target` in any PZ workout. 

<img width="459" alt="Capture" src="https://user-images.githubusercontent.com/52180063/101275222-f66b6e80-3771-11eb-8576-9bb683dbab41.PNG">

**Comment by philosowaffle** [comment]
Interesting. Can you share the Garmin TCX file for that workout with me?  

In my research so far, I have found that when you export the TCX file from garmin it is always missing this extra metadata information which is consistent with my research that the TCX definition does not properly support this extra powerzone data.  

I suspect that syncmyworkout may be writing FIT files instead of TCX.  I've begun a little bit of research on switching over to FIT, but its a much larger project.

**Comment by clipin** [comment]
Yep, [here's workout link](https://connect.garmin.com/modern/activity/5837503646). Zip attached w/ tcx and fit files.
[5837503646.zip](https://github.com/philosowaffle/peloton-to-garmin/files/5649055/5837503646.zip)

**Comment by philosowaffle** [comment]
Yup, confirmed the TCX file from your workout does not have any TargetZone or PowerZone info in it. Looks like I'll be forced to go down the path of learning how FIT files work, because now that I know this data exists... I want it 😛

**Comment by grizzls** [comment]
Would moving to FIT files also allow for the merging of watch data and Peloton data so that we can track the training status and VO2 Max in a single workout like syncmyworkout does but without handing over passwords to a 3rd-party?

**Comment by philosowaffle** [comment]
#57

**Comment by philosowaffle** [comment]
@grizzls - Moving to FIT would at the very least be a pre-requisite to merging this data.  If this is something you're interested in, could you open a new issue and give me some details on the use case and rules for this data?  I don't use Training Status or VO2 max for my cycling work, so I just need some extra insight from someone who does to ensure I understand the feature you're asking for.

Additionally, if you could attach a sample FIT file of the merged data + the original watch FIT file (or link to the strava workout), that would be helpful.

**Comment by philosowaffle** [comment]
Basic support for this has been added as part of P2G V2.  It does have some known limitations, but at a minimum it accurately reflects the time spent in each power zone according to your Peloton FTP at the time of the workout.  Your Peloton FTP is also captured on Garmin Connect with the workout, so you can calculate the exact zone watt values if needed.

This feature can be enabled in V2 by setting "[IncludeTimeInPowerZones](https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration#format-config):" `true`
